File: code/jvm/im-http-api/src/main/kotlin/im/api/model/output/invitations/ImInvitationOutputModel.kt
package im.api.model.output.invitations

import im.domain.invitations.ImInvitation

data class ImInvitationOutputModel(
    val token: String,
    val status: String,
    val expiresAt: String,
) {
    companion object {
        fun fromDomain(invitation: ImInvitation): ImInvitationOutputModel =
            ImInvitationOutputModel(
                token = invitation.token.toString(),
                status = invitation.status.name,
                expiresAt = invitation.expiresAt.toString(),
            )
    }
}

File: code/js/src/Dto/output/invitations/ImInvitationOutputModel.ts
import { ImInvitationStatus } from '../../../Domain/invitations/ImInvitationStatus';

/**
 * Interface for invitation output model
 *
 * @param token The token of the invitation.
 * @param status The status of the invitation.
 * @param expiresAt The date and time when the invitation expires.
 */
export interface ImInvitationOutputModel {
    token: string;
    status: ImInvitationStatus;
    expiresAt: string;
}
